src/main/java/org/venice/beachfront/bfapi/geoserver/GeoserverEnvironment.java

```diff
@@ -0,0 +1,275 @@
+			} 
+			catch (final IOException exception) {
+				String error = String.format(
+						"Received a non-error response from GeoServer resource check for %s, but it was not valid JSON. Authentication may have failed. ",
+						resourceUri, response.getBody());
+				LOG.error(error, exception);
+
+				return false;
+			}
+
+			if (response.getStatusCode().is2xxSuccessful()) {
+				return true;
+			}
+		} 
+		catch (final HttpClientErrorException | HttpServerErrorException exception) {
+			// If it's a 404, then it does not exist. Fall through.
+			if (!exception.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
+				// If it's anything but a 404, then it's a server error and we should not proceed with creation. Throw
+				// an exception.
+				LOG.error("HTTP Status Error checking GeoServer Resource {} Exists : {}" + resourceUri,
+						exception.getStatusCode().toString(), exception);
+			}
+		} 
+		catch (final RestClientException exception) {
+			LOG.error("Unexpected Error Checking GeoServer Resource Exists : " + resourceUri, exception);
+		}
+
+		return false;
+	}
+
+	private void installLayer() {
+
+		authHeaders.setContentType(MediaType.APPLICATION_XML);
+		final String layerURL = String.format("%s/geoserver/rest/workspaces/%s/datastores/%s/featuretypes", GEOSERVER_HOST, WORKSPACE_NAME, DATASTORE_NAME);
+		final HttpEntity<String> request = new HttpEntity<>(getLayerCreationPayload(), authHeaders.get());
+
+		try {
+			restTemplate.exchange(layerURL, HttpMethod.POST, request, String.class);
+
+			LOG.info("GeoServer Layer created successfully!");
+		}
+		catch (final HttpClientErrorException | HttpServerErrorException exception) {
+			final String error = String.format("HTTP Error occurred while trying to create Beachfront GeoServer Layer: %s",
+					exception.getResponseBodyAsString());
+			LOG.error(error, exception);
+		} 
+		catch (final RestClientException exception) {
+			LOG.error("Unexpected Error Creating GeoServer Layer!", exception);
+		}
+	}
+
+	private void installStyle() {
+
+		authHeaders.setContentType(MediaType.parseMediaType("application/vnd.ogc.sld+xml"));
+		final String styleURL = String.format("%s/geoserver/rest/styles?name=%s", GEOSERVER_HOST, STYLE_NAME);
+		final HttpEntity<String> request = new HttpEntity<>(getStyleCreationPayload(), authHeaders.get());
+
+		try {
+			restTemplate.exchange(styleURL, HttpMethod.POST, request, String.class);
+
+			LOG.info("GeoServer Style created successfully!");
+		}
+		catch (final HttpClientErrorException | HttpServerErrorException exception) {
+			final String error = String.format("HTTP Error occurred while trying to create Beachfront GeoServer Style: %s",
+					exception.getResponseBodyAsString());
+			LOG.error(error, exception);
+		} 
+		catch (final RestClientException exception) {
+			LOG.error("Unexpected Error Creating GeoServer Style!", exception);
+		}
+	}
+
+	private String getLayerCreationPayload() {
+
+		return
+			"                <featureType>\n" + 
```

Comment:
Can we create this as an XML file in the src/main/resources folder? I think that might make more sense. We'll have to figure out how to intelligently do string replacemnts (for inserting credentials) but it might be easier to manage this payload as an XML file , if things ever change in the future.

Reply:
Done.